### ISSUE TITLE
Allow configuration by OS environment variables

### ISSUE THREAD

**Issue by dincho** [aeternity/aeternity#3298]
Currently the node can be configuration by command line parameters and configuration file. Where the configuration file itself can be changes by command line parameter or `AETERNITY_CONFIG` OS environment var.

see https://github.com/aeternity/aeternity/blob/master/docs/configuration.md#user-provided-configuration

In the world of containerisation is much more "natural" to use OS environment variables to fully configuration a given piece of software, that would easy the deployment in such environments.

e.g. `AETERNITY_NETWORK_ID`, TBD the exact structure/format

**Comment by uwiger** [comment]
An implementation note: There is already a function in `aeu_env` which _almost_ does this:
```erlang
config_value(CfgKey, App, Env, Default) ->
    {ok, Value} =find_config(CfgKey, [ user_config
                                     , {env, App, Env}
                                     , schema_default
                                     , {value, Default} ]),
    Value.

find_config(CfgKey, [H|T]) ->
    case find_config_(CfgKey, H) of
        undefined -> find_config(CfgKey, T);
        {ok,_} = Ok -> Ok
    end;
find_config(_, []) ->
    undefined.

find_config_(K, user_config       ) -> user_map(K);
find_config_(_, {env, App, EnvKey}) -> get_env(App, EnvKey);
find_config_(K, schema_default    ) -> default(K);
find_config_(_, {value, V}        ) -> {ok, V}.
```
The `get_env/2` function calls `setup:get_env/2`, which [actually (indirectly) supports OS env expansion](https://github.com/uwiger/setup/blob/master/doc/setup.md#variable-expansion), but it would likely be better to add an `{os_env, ...}` option.

**Comment by uwiger** [comment]
Solved in PR #3402 
Closing
